### DISCUSSION TITLE
fontfile - wasm.

### DISCUSSION THREAD

**Discussion by Suganesh-repo** [rdkit/rdkit#7478] [Q&A]
There is a drawing option fontfile, where we can give the path of the .ttf format font file.
Since wasm cannot access local storage files, how does it read ttf file and produce the expected output..

**Comment by DavidACosgrove** [comment]
The default font is built-in at compile time. You can over-ride it with a different .ttf file at run time though probably not via the wasm if it can’t read a local file.

**Comment by flange-ipb** [comment]
> Since wasm cannot access local storage files

That's not completely correct. Emscripten offers [file system support](https://emscripten.org/docs/api_reference/Filesystem-API.html), in particular an in-memory file system inside the browser. With a bit of JavaScript it is possible to upload/download files to/from this file system.
Unfortunately, RDKit.js doesn't seem to build with FS support or it doesn't expose the `FS` object in its API.
